### PR TITLE
slcmd and integration test

### PR DESCRIPTION
**Pull request by zhanggbj**
Hi,

sl --packages and --package-options command need code change in same files, and at the same time, they belong to different stories,  so I submitted two commits in one PR. Thanks for your review:)

[#115478065] {https://www.pivotaltracker.com/story/show/115478065}
[#115478105] {https://www.pivotaltracker.com/story/show/115478105}

### CODE REVIEW

File: cmds/bmp/sl_command.go [hunk 1]
```diff
@@ -45,10 +50,102 @@ func (cmd slCommand) Options() cmds.Options {
 
 func (cmd slCommand) Validate() (bool, error) {
 	cmd.printer.Printf("Validating %s command: options: %#v", cmd.Name(), cmd.options)
+
+	if !cmd.options.Packages && cmd.options.PackageOptions == "" {
```
Comment (maximilien) [inline cmds/bmp/sl_command.go:54]: Hmm, why is cmd.options.Packages now a string?

So when passed you will list all packages?

File: cmds/bmp/sl_command.go [hunk 2]
```diff
@@ -45,10 +50,102 @@ func (cmd slCommand) Options() cmds.Options {
...
+
+		for _, value := range content {
+			table.Append(value)
+		}
+
+		cmd.ui.PrintTable(table)
+	}
+
+	if len(slPackageOptionsResponse.Data.Datacenter) > 0 {
+		cmd.ui.Printf("Packege %s is available in below datacenters:", packageOptions)
```
Comment (maximilien) [inline cmds/bmp/sl_command.go]: Typo: "Packege" should be "Package"

File: cmds/bmp/sl_command.go [hunk 3]
```diff
@@ -45,10 +50,102 @@ func (cmd slCommand) Options() cmds.Options {
...
+
+	if slPackageOptionsResponse.Status != 200 {
+		return slPackageOptionsResponse.Status, nil
+	}
+
+	for _, category := range slPackageOptionsResponse.Data.Category {
+		cmd.ui.Printf("Category Code: %s, Name: %s, Required: %t", category.Code, category.Name, category.Required)
+		cmd.ui.Println("")
+
+		table := tablewriter.NewWriter(os.Stdout)
```
Comment (maximilien) [inline cmds/bmp/sl_command.go:124]: See previous issue I mentioned. I don't think we should be passing os.Stdout. It should come from cmd.Ui. Unless you open a chore to fix this later you need to fix this with the others. Thx.

File: cmds/bmp/sl_command_test.go [hunk 4]
```diff
@@ -65,18 +66,183 @@ var _ = Describe("sl command", func() {
...
+
+			Context("when --package-options specified", func() {
+				BeforeEach(func() {
+					options = cmds.Options{
+						Verbose:        false,
+						PackageOptions: "1",
+					}
+				})
+
+				It("validation pass without errors", func() {
```
Comment (maximilien) [inline cmds/bmp/sl_command_test.go]: This needs in correct English, so, this is better:

'''
It("passes validation without errors", ...)
'''

File: cmds/bmp/sl_command_test.go [hunk 5]
```diff
@@ -65,18 +66,183 @@ var _ = Describe("sl command", func() {
...
+
+			Context("when --package-options specified but no value", func() {
+				BeforeEach(func() {
+					options = cmds.Options{
+						Verbose:        false,
+						PackageOptions: "",
+					}
+				})
+
+				It("validation fails with errors", func() {
```
Comment (maximilien) [inline cmds/bmp/sl_command_test.go]: It("fails validation with errors", ...)

**Comment by maximilien** [comment]
Looks better, however, various small issues to address. Also, you have a file in the commit that needs to be deleted: cmds/bmp/tmp.backup.

Please address all (create chore if you want to separate the table os.Stdout feedback) and I will review quickly and hopefully merge.

File: cmds/bmp/sl_command.go (hunk 1 above)
Comment (zhanggbj) [inline cmds/bmp/sl_command.go:54]: I changed cmd.options.Packages from string to bool, so when execute "bmp sl --packages", all packages will be listed.

File: cmds/bmp/sl_command.go (hunk 3 above)
Comment (zhanggbj) [inline cmds/bmp/sl_command.go:124]: I opened a chore #119336713 to fix this, as the code change about moving this to cmd.ui is in PR #38, so I didn't fix it here.

File: cmds/bmp/sl_command.go (hunk 2 above)
Comment (zhanggbj) [inline cmds/bmp/sl_command.go]: Sorry for my carelessness. Done.

File: cmds/bmp/sl_command_test.go (hunk 4 above)
Comment (zhanggbj) [inline cmds/bmp/sl_command_test.go]: Done.Thanks.

File: cmds/bmp/sl_command_test.go (hunk 5 above)
Comment (zhanggbj) [inline cmds/bmp/sl_command_test.go]: Done

**Comment by zhanggbj** [comment]
Sorry for my carelessness. This is ready for your review. For the comment about command input and output, I opened a chore [#119336713](https://www.pivotaltracker.com/story/show/119336713) and will update all of them together in one PR. Thanks for your review:)

File: cmds/bmp/sl_command.go (hunk 1 above)
Comment (maximilien) [inline cmds/bmp/sl_command.go:54]: OK cool. I'll assume that's the behavior Jimmy et al. are looking for.

File: cmds/bmp/sl_command.go [hunk 6]
```diff
@@ -45,10 +50,102 @@ func (cmd slCommand) Options() cmds.Options {
...
+	slPackagesResponse, err := cmd.bmpClient.SlPackages()
+	if err != nil {
+		return 1, err
+	}
+
+	if slPackagesResponse.Status != 200 {
+		return slPackagesResponse.Status, nil
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
```
Comment (maximilien) [inline cmds/bmp/sl_command.go:89]: You need to switch this to using the `cmd.UI`. I'll accept but please change.

**Comment by maximilien** [comment]
Cool. Merging. Cheers 👍

**Pull request merged**